Matrix.py

```diff
@@ -0,0 +1,120 @@
+class Matrix:
+    # in this program Matrix is a tuple of tuples and the inside tuples must be as long as the major ine is.
+    # data is the Matrix (tuple) received from the user
+    def __init__(self, data):
+        counter = 0
+        if type(data) is tuple:
+            for tup in data:
+                if len(tup) == len(data) and type(tup) is tuple:
+                    counter += 1
+        if counter == len(data):
+            self.data = data
+        else:
+            raise ValueError("Invalid matrix")
+
+    def __iter__(self):
+        return iter(self.data)
+
+    def __getitem__(self, item):
+        return self.data[item]
+
+    def __repr__(self):
+        return f"Matrix: {self.data}"
+
+    # str returns the Matrix in the mathemathic format
+    def __str__(self):
+        matrix_in_lines = ""
+        for i in range(len(self.data)):
+            matrix_in_lines += (str(self.data[i]) + "\n")
+        return matrix_in_lines
+
+    def get_tuple(self):
+        return self.data
+
+    # unity method creates a matrix of zeros with a diagonal of ones (unity is the Matrix's size)
+    @staticmethod
+    def unity(unit):
+        major_list = [[1 if i == index else 0 for i in range(unit)] for index in range(unit)]
```

Comment:
add a comment that describes what you are doing in this list comprehension. it will be easier for us and for you in the future to understand it.

* do this in all of your complex list comprehensions